woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php

```diff
@@ -141,65 +141,39 @@
 	);
 
 	/**
-	 * This function splits an address line like for example "Pallaswiesenstr. 45 App 231" into its individual parts.
-	 * Supported parts are additionToAddress1, streetName, houseNumber and additionToAddress2. AdditionToAddress1
-	 * and additionToAddress2 contain additional information that is given at the start and the end of the string, respectively.
-	 * Unit tests for testing the regular expression that this function uses exist over at https://regex101.com/r/vO5fY7/1.
-	 * More information on this functionality can be found at http://blog.viison.com/post/115849166487/shopware-5-from-a-technical-point-of-view#address-splitting.
+	 * Fatia um endereço em nome da rua, número e complemento
 	 *
-	 * @param string $address
+	 * @param  string $address Endereço a ser fatiado
 	 * @return array
 	 */
-	public static function split_street($address)
-	{
-		$regex = '/\A\s*
-		   (?: #########################################################################
-			   # Option A: [<Addition to address 1>] <House number> <Street name>      #
-			   # [<Addition to address 2>]                                             #
-			   #########################################################################
-			   (?:(?P<A_Addition_to_address_1>.*?),\s*)? # Addition to address 1
-		   (?:No\.\s*)?
-			   (?P<A_House_number>\pN+[a-zA-Z]?(?:\s*[-\/\pP]\s*\pN+[a-zA-Z]?)*) # House number
-		   \s*,?\s*
-			   (?P<A_Street_name>(?:[a-zA-Z]\s*|\pN\pL{2,}\s\pL)\S[^,#]*?(?<!\s)) # Street name
-		   \s*(?:(?:[,\/]|(?=\#))\s*(?!\s*No\.)
-			   (?P<A_Addition_to_address_2>(?!\s).*?))? # Addition to address 2
-		   |   #########################################################################
-			   # Option B: [<Addition to address 1>] <Street name> <House number>      #
-			   # [<Addition to address 2>]                                             #
-			   #########################################################################
-			   (?:(?P<B_Addition_to_address_1>.*?),\s*(?=.*[,\/]))? # Addition to address 1
-			   (?!\s*No\.)(?P<B_Street_name>[^0-9# ]\s*\S(?:[^,#](?!\b\pN+\s))*?(?<!\s)) # Street name
-		   \s*[\/,]?\s*(?:\sNo[.:])?\s*
-			   (?P<B_House_number>\pN+\s*-?[a-zA-Z]?(?:\s*[-\/\pP]?\s*\pN+(?:\s*[\-a-zA-Z])?)*|
-			   [IVXLCDM]+(?!.*\b\pN+\b))(?<!\s) # House number
-		   \s*(?:(?:[,\/]|(?=\#)|\s)\s*(?!\s*No\.)\s*
-			   (?P<B_Addition_to_address_2>(?!\s).*?))? # Addition to address 2
-		   )
-		   \s*\Z/xu';
-
-		$result = preg_match($regex, $address, $matches);
+	public static function split_street($address) {
```

Comment:
That regex was not working when address had no number... a merchant added an address number field to their checkout (using checkout manager) and that broke things as address had no number and regex simply failed...
As it was too complex we rewrote the regex to be simplier and to match with weird addresses anyway.